### PR TITLE
[PM-8115] Desktop, Extension UI Refresh: Self-hosted Setup Dialog

### PR DESCRIPTION
**Pull request by alec-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-8115?atlOrigin=eyJpIjoiZjYxZDhiNjg1OWQwNDFlZGI3NWQyYmVkYTZlMGVhNGYiLCJwIjoiaiJ9

## 📔 Objective

Uses the new `RegistrationSelfHostedEnvConfigDialogComponent` as the dialog when selecting "self-hosted" from the server configuration on the login page. Also adjusts the leading text for the select based on whether the `UnauthenticatedExtensionUIRefresh` is enabled.

## 📸 Screenshots

Screenshot 2024-10-16 at 9 28 34 PM
Screenshot 2024-10-16 at 9 28 24 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/browser/src/auth/popup/home.component.ts
Comment (JaredSnider-Bitwarden) [inline apps/browser/src/auth/popup/home.component.ts:1]: I apologize that the task was not clear enough on this, but we do not need to modify the existing login components - only the new login component's env selector.

Comment (alec-livefront) [reply apps/browser/src/auth/popup/home.component.ts:1]: @JaredSnider-Bitwarden does this mean this needs to branch off of 8111?

Comment (JaredSnider-Bitwarden) [reply apps/browser/src/auth/popup/home.component.ts:1]: No, upon reviewing the changesets, this can be done separately.

**Review by JaredSnider-Bitwarden — CHANGES_REQUESTED** [review]
I tested this really quickly, but the self hosted option doesn't open on desktop or browser after merging main? 

https://github.com/user-attachments/assets/31a1648a-47dc-4afc-b98f-b60f33618463

**Comment by alec-livefront** [comment]
@JaredSnider-Bitwarden I believe I've implemented everything we talked about this morning:

- Moving / renaming the `SelfHostedEnvConfigDialogComponent`
- Removing the EnvironmentSelectorService
- Removing the instance of "loggingInOn" from browser messages

File: libs/angular/src/auth/components/environment-selector.component.ts [hunk 1]
```diff
@@ -102,8 +109,19 @@ export class EnvironmentSelectorComponent implements OnInit, OnDestroy {
...
+     * Opens the self-hosted settings dialog.
+     *
+     * If the `UnauthenticatedExtensionUIRefresh` feature flag is enabled,
+     * the self-hosted settings dialog is opened directly. Otherwise, the
+     * `onOpenSelfHostedSettings` event is emitted.
+     */
     if (option === Region.SelfHosted) {
-      this.onOpenSelfHostedSettings.emit();
+      if (await this.configService.getFeatureFlag(FeatureFlag.UnauthenticatedExtensionUIRefresh)) {
+        await SelfHostedEnvConfigDialogComponent.open(this.dialogService);
```
Comment (JaredSnider-Bitwarden) [inline libs/angular/src/auth/components/environment-selector.component.ts:120]: This is looking a lot better! We should add a success toast if the return result is successful of:
```TypeScript
  this.toastService.showToast({
        variant: "success",
        title: null,
        message: this.i18nService.t("environmentSaved"),
      });
```
Taken from the `RegistrationEnvSelectorComponent`.  "environmentSaved" already exists on each client as well.

File: libs/angular/src/auth/components/environment-selector.component.ts [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+import { FeatureFlag } from "@bitwarden/common/enums/feature-flag.enum";
+import { ConfigService } from "@bitwarden/common/platform/abstractions/config/config.service";
 import {
   EnvironmentService,
   Region,
   RegionConfig,
 } from "@bitwarden/common/platform/abstractions/environment.service";
+import { DialogService } from "@bitwarden/components";
+
+import { SelfHostedEnvConfigDialogComponent } from "../../../../auth/src/angular/self-hosted-env-config-dialog/self-hosted-env-config-dialog.component";
```
Comment (JaredSnider-Bitwarden) [inline libs/angular/src/auth/components/environment-selector.component.ts]: ⚠️ , anytime you see an import like this from our auth directory, it should be a red flag to indicate that we are missing an `index.ts` export somewhere. Anything coming out of our libs/auth should be either imported from `from "@bitwarden/auth/angular"` or `from "@bitwarden/auth/common"`

Comment (JaredSnider-Bitwarden) [reply libs/angular/src/auth/components/environment-selector.component.ts]: Here's the way to resolve this: https://github.com/bitwarden/clients/pull/11597/commits/a8cc5df806bc7d4fa8081d735f24306066084015

**Review by JaredSnider-Bitwarden — CHANGES_REQUESTED** [review]
Nice work! This is incredibly close! Just a few notes below:

File: libs/angular/src/auth/components/environment-selector.component.ts (hunk 1 above)
Comment (alec-livefront) [reply libs/angular/src/auth/components/environment-selector.component.ts]: [7cc875f](https://github.com/bitwarden/clients/pull/11597/commits/7cc875ff336dbd98bbbf35549f204b7c2afd1770)

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LGTM!

**Pull request merged**